Various updates to "Customization" (#1358)
semgrep/semgrep-docs: docs/semgrep-supply-chain/getting-started.md

## Scan frequency

By adjusting your CI configuration, you can configure your scans so that Semgrep Supply Chain scans your code at a different frequency or when a specific event occurs.

### Schedule scans

Semgrep Supply Chain frequently receives rule updates. To take advantage of these updates, adjust the frequency with which Semgrep Supply Chain scans your codebase.

<CiScheduling />

:::note Rules updates

> If a rule is updated, findings generated against the updated rule are considered **new findings**, even if the previous version of the rule generated a finding. Because the finding is new, you'll receive notifications through the channels you've set up (e.g., Slack, email).

Author: @andy-r2c This line had stuck with me since I made a wording change unrelated to the content, but based on separate discussion it sounds like it might not even be correct? Could you confirm?

## Scan a monorepo's dependencies

Semgrep Supply Chain supports the scanning of monorepos. As outlined in [Project directory structure](#project-directory-structure), findings are grouped by directory based on the [lockfiles](/semgrep-supply-chain/glossary/#lockfile) or manifest files present in the monorepo.

## Block pull or merge requests

> Semgrep Supply Chain versions **v0.122.0** and earlier automatically blocked pull/merge requests if it discovered reachable findings in the code, but later versions do not do this. You can, however, configure Semgrep Supply Chain to block on pull request scans that detect reachable findings with high or critical severity.

Reviewer: Slight subtlety: we will block on any reachable code finding with these severities, but we will only block on a lockfile-only "always reachable" finding if it is high/critical and it's a _direct_ dependency. Always reachable findings on transitive dependencies won't block, because we thought it would be too noisy.

Reviewer: It depends whether we consider it worth it to bring up the "directly importing a transitive dependency" edge case here. The blocking situation is this:

 |  | Reachable in code      | Always reachable in lockfile |
| ----------- | ----------- | ----------- |
| Direct dep      | Block       | Block |
| Transitive dep  | Block*        | No Block |

\* Only happens in edge case that should be avoided

Feels hard to keep it snappy 😅

Author: Yep - I'm not sure we need to cover the weird edge case in the glossary, but good point that we might want to consider covering it elsewhere - it might be a good fit for the KB.

> Semgrep Supply Chain with Semgrep versions **v0.122.0** and earlier automatically blocked pull/merge requests if it discovered reachable findings in the code, but more recent versions of Semgrep do not do this. You can, however, configure Semgrep Supply Chain to block on pull request scans that detect reachable findings with high or critical severity.

Reviewer 2: Would this make sense? 

```suggestion
Semgrep Supply Chain versions **v0.122.0** and earlier automatically blocked pull/merge requests if it discovered reachable findings in the code, but later versions do not do this. You can, however, configure Semgrep Supply Chain to block on pull request scans that detect reachable findings with high or critical severity.
```
